MapboxVision/src/main/java/com/mapbox/vision/video/videosource/camera/CameraVideoSourceImpl.kt

```diff
@@ -328,96 +249,78 @@ internal class CameraVideoSourceImpl(
+                if (characteristics.get(CameraCharacteristics.LENS_FACING) == CameraCharacteristics.LENS_FACING_FRONT) {
                     continue
                 }
 
-                val map = characteristics.get(CameraCharacteristics.SCALER_STREAM_CONFIGURATION_MAP)
-                        ?: continue
+                val sizes = characteristics
+                                    .get(CameraCharacteristics.SCALER_STREAM_CONFIGURATION_MAP)
+                                    ?.getOutputSizes(SurfaceTexture::class.java)
+                                    ?.toList()
+                            ?: continue
 
-                previewSize = chooseOptimalSize(
-                        choices = map.getOutputSizes(SurfaceTexture::class.java),
-                        width = width,
-                        height = height
+                previewSize = chooseOptimalCameraResolution(
+                        supportedSizes = sizes,
+                        desiredSize = Size(desiredWidth, desiredHeight)
                 )
-
-                /* Orientation of the camera sensor */
                 sensorOrientation = characteristics.get(CameraCharacteristics.SENSOR_ORIENTATION)!!
 
-                val focalLengths = characteristics.get(CameraCharacteristics.LENS_INFO_AVAILABLE_FOCAL_LENGTHS)
-                val focalLength = if (focalLengths != null && focalLengths.isNotEmpty()) {
-                    focalLengths[0]
-                } else {
-                    0f
-                }
-                val sensorSize = characteristics.get(CameraCharacteristics.SENSOR_INFO_PHYSICAL_SIZE)
-
-                rgbaBytes = ByteArray(previewSize.width * previewSize.height * 4)
-                rgbBitmap = Bitmap.createBitmap(previewSize.width, previewSize.height, Bitmap.Config.ARGB_8888)
+                val focalLength = characteristics
+                        .get(CameraCharacteristics.LENS_INFO_AVAILABLE_FOCAL_LENGTHS)!!
+                        .first()
 
-                cameraParamsRequestsManager = CameraParamsRequestsManager(previewSize, focalLength, sensorSize)
-                        .also {
-                            it.cameraParamsListener = object : CameraParamsListener {
-                                override fun onCameraParamsReady(cameraParamsData: CameraParamsData) {
-                                    videoSourceListener?.onNewCameraParams(cameraParamsData)
-                                }
-                            }
-                        }
+                val physicalSensorSize = characteristics.get(CameraCharacteristics.SENSOR_INFO_PHYSICAL_SIZE)!!
+                val focalInPixelsX: Float = focalLength * previewSize.width / physicalSensorSize.width
+                val focalInPixelsY: Float = focalLength * previewSize.height / physicalSensorSize.height
 
-                this.cameraId = cameraId
+                cameraParams = CameraParamsData(
```

Comment:
Parameters are now counted here once and updated when listener attaches in `attach`